ygen/genstate.go

```diff
@@ -108,15 +108,15 @@ func newGenState() *genState {
 // this function in common with enumerations that are not within a union. The name of the enumerated
 // value is calculated based on the original context, and whether path compression is enabled based
 // on the compressPaths boolean.
-func (s *genState) enumeratedUnionEntry(e *yang.Entry, compressPaths bool) ([]*yangEnum, error) {
+func (s *genState) enumeratedUnionEntry(e *yang.Entry, compressPaths, omitUnderscores bool) ([]*yangEnum, error) {
```

Comment:
Some explanation of the reason behind omitUnderscores would be useful. For example why not have the positive version "underscores" and omit all the negation? 
If sticking with negation perhaps change to noUnderscores?

Reply:
Added a comment re: languages that don't allow this in the styles. I prefer the names with underscores in, hence the preference for this to be on.